_ingest-pipelines/processors/trim.md

```diff
@@ -0,0 +1,166 @@
+---
+layout: default
+title: Trim
+parent: Ingest processors
+nav_order: 300
+---
+
+# Trim processor
+
+The `trim` processor is used to remove leading and trailing white space characters from a specified field.
+
+The following is the syntax for the `trim` processor:
+
+```json
+{
+  "trim": {
+    "field": "field_to_trim",
+    "target_field": "trimmed_field"
+  }
+}
+```
+{% include copy-curl.html %}
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `trim` processor.
+
+Parameter | Required/Optional | Description |
+|-----------|-----------|-----------|
+`field` | Required | The field containing the text to be trimmed. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets).
+`target_field` | Required | The field where the trimmed text is stored. If not specified, the trimmed text is stored in the same field as the original text. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets).
```

Comment:
`target_field` is optional and doesn't support template snippets. By default, `field` is updated in place.